xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -341,6 +342,42 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
       );
   }
 }
+
+/**
+ * Event listener, called when clicking the "delete all" button.
+ *
+ * @param event the link activation event
+ */
+XWiki.index.trash.documents.confirmDeleteAll = function(event) {
+  event.stop();
+  if (this.hasClassName('disabled')) {
+    // Do nothing if the button was already clicked and it's waiting for a response from the server.
+    return;
+  } else {
+    new XWiki.widgets.ConfirmedAjaxRequest(
+      /* Ajax request URL */
+      this.href = "$xwiki.getURL($doc.fullName, 'delete', '&amp;confirm=1&amp;emptybin=true')" + "&amp;form_token=$!{services.csrf.getToken()}" + (Prototype.Browser.Opera ? "" : "&amp;ajax=1"),
```

Comment:
I don't understand this syntax. You do an assignment and the result of evaluating the assignment is passed as the first parameter to the ``ConfirmedAjaxRequest`` constructor? ``this`` must be the link. Why do you need to set the link ``href`` property?

Note that if you need to get the URL of the current document in JavaScript you don't need Velocity. You can do

    XWiki.currentDocument.getURL('delete')

Also, the line is too long and not that easy to read. I would pass the request parameters using

    ``parameters: {}``

below. See http://api.prototypejs.org/ajax/ .

Reply:
> Why do you need to set the link href property?

Because right now when I build the link, the href property only contains `#`, which is wrong: I should put the right URL using:
`XWiki.currentDocument.getURL('delete')`. I'll change that.

Then I'll still need to add the confirm parameter at that place.

Reply:
> Because right now when I build the link, the href property only contains #, which is wrong

Why is it wrong? Why does it matter the value of the link ``href`` attribute? You're stopping the click event and you're doing an AJAX request. The link is not followed.

Reply:
> Why is it wrong?

I was thinking about the case when javascript is not available, but anyway the button won't be available in JS is deactivated so indeed it doesn't change anything.